From 8cd778650ae223cd306588042b55d0290ef81037 Mon Sep 17 00:00:00 2001
Message-Id: <8cd778650ae223cd306588042b55d0290ef81037.1649937053.git.geliang.tang@suse.com>
From: Kelvin Cao <kelvin.cao@microchip.com>
Date: Thu, 23 Dec 2021 17:23:34 -0800
Subject: [PATCH] ntb_hw_switchtec: Fix a minor issue in config_req_id_table()
Git-commit: 8cd778650ae223cd306588042b55d0290ef81037
Patch-mainline: v5.17-rc1
References: bsn#170
Modified-by-SEL: No


The req_id_table_size field is 16-bit wide, use ioread16() to read the
value.

Signed-off-by: Kelvin Cao <kelvin.cao@microchip.com>
Signed-off-by: Jon Mason <jdmason@kudzu.us>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/ntb/hw/mscc/ntb_hw_switchtec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/ntb/hw/mscc/ntb_hw_switchtec.c b/drivers/ntb/hw/mscc/ntb_hw_switchtec.c
index e79a355bafbf..88ae18b0efa8 100644
--- a/drivers/ntb/hw/mscc/ntb_hw_switchtec.c
+++ b/drivers/ntb/hw/mscc/ntb_hw_switchtec.c
@@ -955,7 +955,7 @@ static int config_req_id_table(struct switchtec_ntb *sndev,
 	u32 error;
 	u32 proxy_id;
 
-	if (ioread32(&mmio_ctrl->req_id_table_size) < count) {
+	if (ioread16(&mmio_ctrl->req_id_table_size) < count) {
 		dev_err(&sndev->stdev->dev,
 			"Not enough requester IDs available.\n");
 		return -EFAULT;
-- 
2.34.1

